src/api/localResolvers/experiment.js

```diff
@@ -139,11 +139,11 @@ export default ({ cookieStore }) => {
 					// get the existing assigned version for this experiment id
 					let version = assignments[id];
 
-					// assign an experiment version if it's currently undefined
-					if (_isUndefined(version)) {
-						// read the experiment data from the cache
-						const experiment = readJSONSetting(context, `cache.data.data['Setting:uiexp.${id}'].value`);
+					// read the experiment data from the cache
+					const experiment = readJSONSetting(context, `cache.data.data['Setting:uiexp.${id}'].value`);
 
+					// assign an experiment version if it's currently undefined
+					if (experiment && _isUndefined(version)) {
```

Comment:
also added this check for cases when the exp may not be in the cache when we look for it.